### DISCUSSION TITLE
Restart i3 and focus issue

### DISCUSSION THREAD

**Discussion by c02y** [i3/i3#4593] [Q&A] [answered]
I set `assign` and `for_window` with `focus`, but every time I restart i3, the focus will be changed to the focused window

the whole config is the /etc/i3/config and the following config
```
assign [class="TelegramDesktop"] $ws2
for_window [class="TelegramDesktop"] focus
```

no matter what `$ws2` is and no matter what workspace and window I'm current using, once I restart i3, the focus will go to `TelegramDesktop` window

FYI:
`TelegramDesktop` is not the only one.

So what I need is:
when I open a programA, it is opened in the specific workspace and change focus to that program, then when I'm in another window such as editing my config, and restart i3, it should stay inside the editor, but now it goes to the programA .

Is this a bug or a feature? How can I avoid this?

**Comment by Airblader** [comment, marked as answer]
> Is this a bug

I don't know why it possibly would be. You're literally telling i3 to do exactly that, so why shouldn't it do it...?

Maybe you don't want to restart i3 in-place (`restart`), but just reload the config (`reload`)?

**Reply by c02y** [reply]
Yeah, I believe I misunderstood the following command before this post
```
for_window [class="xxx"] focus
```

I'd like to use `restart` since I found some configs such as `bar` do not take effect after `reload`, I'm just starting to use i3 for two days.

Anyway, if I remove that line, `restart` will be OK.

But I am not able to open a program to a specific workspace and start to use(focus) it immediately without manual focus.

Is there a simple way to do that?

**Comment by Airblader** [comment]
> since I found some configs such as bar do not take effect after reload

On recent versions of i3 it should IIRC.

> Is there a simple way to do that?

That's what the line you posted does.

**Reply by c02y** [reply]
colors/client sometimes don't work, `for_window`s sometimes don't work, anyway, I will use reload as the first choice. 

As for the focus, I'll stop using `for_window`+`focus` together.

**Reply by orestisfl** [reply]
You will need to be more specific regarding what you try to do and what doesn't work. Also, what's your i3 --version?
